Add resize and zscore

Teschl opened the pull request:
This adds the functions `zscore` and `resize` to the `GridObject`.

Maybe `crop` would be a more fitting name for the `resize` function. Also, having the arguments `left`, `right`, `top` and `bottom` might be unnecessary. The function only needs two x and y values each and figures out which is left/right or top/bottom automatically. Using `highlight_selected` will create a plot which highlights the area the new `GridObject` will be cropped to. 
I am not 100% sure the updating of the bounds and transform is completely correct, so if anything seems off about that please let me know.
Closes #264

wkearn reviewed (changes requested):
`zscore` looks good! Happy to merge this now if you'd like to split it off into another PR, or we can wait until we figure out `resize`.

`resize` works great. There is some subtlety about the transform/bounds calculation that should be addressed, and I have a question about the plots.

Teschl commented:
I implemented the requested changes. I added the plotting a rectangle example to the docstring since I was able to compress it into a relatively small space.
Closes #264
